main.py

```diff
@@ -4,24 +4,96 @@
-channel_table = []
 
 # Whenever a message is sent.
 @client.event
 async def on_message(message):
     # we do not want the bot to reply to itself
     if message.author == client.user:
         return
+    
+    # Check if the message author has the Game Master role
+    isGameMaster = False
+    if False:
+        isGameMaster = True
+
+    result = process(message,isGameMaster)
+
+    temp_msg = []
+
+    gamelog_channel = client.get_channel(config.game_log)
+    botspam_channel = client.get_channel(config.bot_spam)
+    storytime_channel = client.get_channel(config.story_time)
+
+    for mailbox in result:
+
+        for element in mailbox.gamelog:
+            msg = await client.send_message(gamelog_channel,element.content)
+            if element.temporary == True:
+                temp_msg.append(msg)
+
+        for element in mailbox.botspam:
+            msg = await client.send_message(botspam_channel,element.content)
+            if element.temporary == True:
+                temp_msg.append(msg)
+        
+        for element in mailbox.storytime:
+            msg = await client.send_message(storytime_channel,element.content)
+            if element.temporary == True:
+                temp_msg.append(msg)
+        
+        for element in mailbox.channel:
+            msg = await client.send_message(client.get_channel(element.destination),element.content)
+            if element.temporary == True:
+                temp_msg.append(msg)
+        
+        for element in mailbox.player:
+            msg = await client.send_message('''How did we do this again?''',element.content)
+            if element.temporary == True:
+                temp_msg.append(msg)
+        
+        for element in mailbox.oldchannels:
+            # element.channel - channel to be edited;
+            # element.victim - person's permission to be changed;
+            # element.number - type of setting to set to:
+                # 0 - no access     (no view, no type)
+                # 1 - access        (view + type)
+                # 2 - frozen        (view, no type)
+                # 3 - abducted      (no view, no type)
+                # 4 - dead          (dead role?)
+            
+            # TODO
+            pass
+
+        for element in mailbox.newchannels:
+            # element.name - name of the channel;
+            # element.owner - owner of the channel;
+            #
+            # @Participant      - no view + type
+            # @dead Participant - view + no type
+            # @everyone         - no view + no type
+
+            # All you need to do is create a channel where only the channel owner has access.
+            # The other members are given access through another Mailbox.
+            # You could make the work easier if you also posted a cc channel message already over here.
+
+            # TODO
+
+            for buddy in element.settlers:
+                db_set(buddy,"channel",'''id of the channel you just created''')
```

Comment:
wtf is this?

Reply:
THAT code is used to set players' personal channels at the beginning of the game.

You know, so that we know when players can or cannot use their abilities. To prevent accidental spoilers.